src/fstab-generator/fstab-generator.c

```diff
@@ -635,9 +635,17 @@ static int parse_fstab(bool initrd) {
                         k = chase_symlinks(where, initrd ? "/sysroot" : NULL,
                                            CHASE_PREFIX_ROOT | CHASE_NONEXISTENT,
                                            &canonical_where, NULL);
-                        if (k < 0) /* If we can't canonicalize, continue as if it wasn't a symlink */
-                                log_debug_errno(k, "Failed to read symlink target for %s, ignoring: %m", where);
-                        else if (streq(canonical_where, where)) /* If it was fully canonicalized, suppress the change */
+                        if (k < 0) {
+                                /* If we can't canonicalize, continue as if it wasn't a symlink */
+                                log_debug_errno(k, "Failed to read symlink target for %s, using as-is: %m", where);
+
+                                if (initrd) {
+                                        canonical_where = path_join("/sysroot", where);
+                                        if (!canonical_where)
+                                                return log_oom();
+                                }
+
+                        } else if (streq(canonical_where, where)) /* If it was fully canonicalized, suppress the change */
```

Comment:
hmm, so chase_symlinks() her already has CHASE_NONEXISTENT set, so the example you gave in the commit msg, since the case /foo/ should already be covered. 

CHASE_NONEXISTENT only allows the basename of the input path to be missing. Maybe we should add CHASE_NONEXISTENT_ANY which allows any suffix to be missing, including one consisting of a multi-component path.

the fix to chase_symlinks() should be pretty easy to allow that, and the overall result should be much nicer

Reply:
(CHASE_NONEXISTANT should probably be renamed CHASE_NONEXISTANT_LAST or so)

Reply:
Yeah, that's an option. But I think we can merge those patches as they are, in particular this has the advantage that they are easy to backport. We can always simplify the code if somebody makes `chase_symlinks()` more powerful.

Reply:
Yeah, that's an option. But I think we can merge those patches as they are, in particular this has the advantage that they are easy to backport. We can always simplify the code if somebody makes `chase_symlinks()` more powerful.